service/history/handler.go

```diff
@@ -1580,42 +1470,120 @@ func (h *Handler) ReapplyEvents(
+	h.GetMetricsClient().IncCounter(scope, metrics.CadenceRequests)
+	sw := h.GetMetricsClient().StartTimer(scope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	engine, err := h.controller.getEngineForShard(int(request.GetShardID()))
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	resp, err := engine.ReadDLQMessages(ctx, request)
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	return resp, nil
 }
 
-// PurgeDLQMessages deletes replication DLQ messages
-func (h *Handler) PurgeDLQMessages(
-	ctx context.Context,
-	request *r.PurgeDLQMessagesRequest,
-) (retError error) {
-	panic("PurgeDLQMessages should not be called for thrift handler")
+func (h *Handler) PurgeDLQMessages(ctx context.Context, request *historyservice.PurgeDLQMessagesRequest) (_ *historyservice.PurgeDLQMessagesResponse, retError error) {
+	defer log.CapturePanicGRPC(h.GetLogger(), &retError)
+
+	h.startWG.Wait()
+
+	scope := metrics.HistoryPurgeDLQMessagesScope
+	h.GetMetricsClient().IncCounter(scope, metrics.CadenceRequests)
+	sw := h.GetMetricsClient().StartTimer(scope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	engine, err := h.controller.getEngineForShard(int(request.GetShardID()))
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	err = engine.PurgeDLQMessages(ctx, request)
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+	return &historyservice.PurgeDLQMessagesResponse{}, nil
 }
 
-// MergeDLQMessages reads and applies replication DLQ messages
-func (h *Handler) MergeDLQMessages(
-	ctx context.Context,
-	request *r.MergeDLQMessagesRequest,
-) (resp *r.MergeDLQMessagesResponse, retError error) {
-	panic("MergeDLQMessages should not be called for thrift handler")
+func (h *Handler) MergeDLQMessages(ctx context.Context, request *historyservice.MergeDLQMessagesRequest) (_ *historyservice.MergeDLQMessagesResponse, retError error) {
+	defer log.CapturePanicGRPC(h.GetLogger(), &retError)
+
+	h.startWG.Wait()
+
+	scope := metrics.HistoryMergeDLQMessagesScope
+	h.GetMetricsClient().IncCounter(scope, metrics.CadenceRequests)
+	sw := h.GetMetricsClient().StartTimer(scope, metrics.CadenceLatency)
+	defer sw.Stop()
+
+	engine, err := h.controller.getEngineForShard(int(request.GetShardID()))
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	resp, err := engine.MergeDLQMessages(ctx, request)
+	if err != nil {
+		err = h.error(err, scope, "", "")
+		return nil, err
+	}
+
+	return resp, nil
 }
 
-// RefreshWorkflowTasks refreshes all the tasks of a workflow
-func (h *Handler) RefreshWorkflowTasks(
-	ctx context.Context,
-	request *hist.RefreshWorkflowTasksRequest) (retError error) {
-	panic("RefreshWorkflowTasks should not be called for thrift handler")
+func (h *Handler) RefreshWorkflowTasks(ctx context.Context, request *historyservice.RefreshWorkflowTasksRequest) (_ *historyservice.RefreshWorkflowTasksResponse, retError error) {
```

Comment:
I already moved this entire API to grpc.  I don't think we need the thrift handler definitions for this.

Reply:
Just making sure you completely removed the thrift handlers for history service and now only handler we register is grpc based?